Firestore/core/src/firebase/firestore/immutable/sorted_set.h

```diff
@@ -74,6 +74,22 @@ class SortedSet : public SortedContainer {
     return SortedSet{map_.insert(key, {})};
   }
 
+  ABSL_MUST_USE_RESULT SortedSet union_with(const SortedSet<K>& other) const {
+    const auto result_ptr = this;
```

Comment:
Optional: since the type isn't onerous here (`const SortedSet*`), consider spelling it out. While explicit types are somewhat annoying they help reduce the cognitive load on the reader.

Reply:
Here actually requires casting, which made `auto` seems OK now.

The PR does not build yet when you reviewed. But I also did not assign it yet, so it's not my fault :-)